### ISSUE TITLE
Set a relative path to cache dir

### ISSUE THREAD

**Issue by dmpetrov** [iterative/dvc#1284]
DVC from sources: 
```
$ dvc -V
0.20.0+c4295f.mod
```

```
$ git clone https://github.com/iterative/so-dataset-posts-25K
$ cd so-dataset-posts-25K/
$ dvc config cache.dir ../mycache
$ dvc pull
....
$ ls mycache # <-- it suppose to be one level up ../mycache
2f
```

Btw... it works just fine if I use an absolute path like in this example `dvc config cache.dir /dvc-cache`: https://dvc.org/doc/use-cases/multiple-data-scientists-on-a-single-machine

**Comment by mroutis** [comment]
Hi, @dmpetrov !

I'm looking at the source code right now, and it looks like the relative paths work, but are relative to the `.dvc` directory, should we make it relative to the working directory?

**Comment by mroutis** [comment]
@dmpetrov , I opened a pull request for this at https://github.com/iterative/dvc/pull/1318

Feel free to close it if you don't think it's necessary to make a change on the current logic.

**Comment by efiop** [comment]
We should just make it clear in docs that if path is relative, it is resolved relative to dvc_dir.

**Comment by dmpetrov** [comment]
Does it means that it will work for an absolute path `dvc config cache.dir /home/dmitry/mycache` but it won't for relative path `dvc config cache.dir ../../mycache`? The behavior should be the same.

**Comment by dmpetrov** [comment]
I agree that we cannot use a relative path from the current directory. However, a relative path from a project directory still sounds to me like a better option (comparing to relative to `.dvc`). Now it is very counterintuitive even if we put it into the documentation.

git-config example might be not very relevant because it is about the internals, not about user level API. And even in Git, user level API work with path relative to project dir:
```
$ git branch some-existing-branch
$ git worktree add ../proj_worktree1 some-existing-branch
Preparing ../proj_worktree1 (identifier proj_worktree1)
HEAD is now at e909e9f2 Update README.md
```

It would be helpful to see some scenarios when a path relative to `.dvc` makes more sense.

**Comment by efiop** [comment]
@dmpetrov But we also will have user and system configs, which will not have project dir to go along with them, so paths in them will have to be specified relative to their location anyway. So in order to provide the same behavior for all configs, we need to keep the current behavior.

**Comment by efiop** [comment]
I've used `worktree` as an example, but there other ones like `include` sections in git config. See https://git-scm.com/docs/git-config#_conditional_includes . And once again, relative paths in them are resolved relative to the config file location, which is correct and expected behavior. And git doesn't introduce a separate command `git include` to do that single action. And I also see no reason to bloat our CLI with yet another command just to provide an implicit behavior, when we already have `dvc config cache.dir` that has consistent behavior on all levels.

**Comment by mroutis** [comment]
I moved the issue to dvc.org to add a better explanation of this behavior on the docs.
